feat/tests/test_fast_detector.py

```diff
@@ -35,3 +44,225 @@ def test_detect_image(self, single_face_img):
 
         # AU checks; TODO: Add more
         assert fex.aus.AU20[0] > 0.8
+
+        
+    def test_fast_landmark_with_batches(self, multiple_images_for_batch_testing):
+        """
+        Make sure that when the same images are passed in with and without batch
+        processing, the detected landmarks and poses come out to be the same
+        """
+        det_result_batch = self.detector.detect(
+            inputs=multiple_images_for_batch_testing,
+            batch_size=5,
+        )
+
+        det_result_no_batch = self.detector.detect(
+            inputs=multiple_images_for_batch_testing,
+            batch_size=1,
+        )
+
+        assert np.allclose(
+            det_result_batch.loc[:, "x_0":"y_67"].to_numpy(),
+            det_result_no_batch.loc[:, "x_0":"y_67"].to_numpy(),
+        )
+            
+    # TODO: Currently making this test always pass even if batching gives slightly diff
+    # results until @tiankang can debug whether we're in tolerance
+    # Track progress updates in this issue: https://github.com/cosanlab/py-feat/issues/128
+    def test_fast_detection_and_batching_with_diff_img_sizes(
+        self, single_face_img, multi_face_img, multiple_images_for_batch_testing
+    ):
+        """
+        Make sure that when the same images are passed in with and without batch
+        processing, the detected landmarks and poses come out to be the same
+        """
+        # Each sublist of images contains different sizes
+        all_images = (
+            [single_face_img] + [multi_face_img] + multiple_images_for_batch_testing
+        )
+
+        # Multiple images with different sizes are ok as long as batch_size == 1
+        # Detections will be done in each image's native resolution
+        det_output_default = self.detector.detect(inputs=all_images, batch_size=1)
+
+        # TODO: fast_detector does not currently raise ValueError
```

Comment:
Here's the fix for this if you want to add it just above the for loop on line 848 of `FastDetector.py`:

```
try:
    _ = next(enumerate(tqdm(data_loader)))
except RuntimeError as e:
    raise ValueError(f"When using `batch_size > 1`, all images must either have the same dimension or `output_size` should be something other than `None` to pad images prior to processing\n{e}")
```

We can just rely on pytorch's check about mismatched image sizes and `ouptut_size=None` and throw our own more friendly error.